Kinvey/KinveyTests/MultiInsertSpec.swift

```diff
@@ -0,0 +1,1600 @@
+                            let result = kinveySaveMulti(dataStore: networkDataStore, entities: Person { $0.name = "Victor" }, Person { $0.name = "Hugo" }).result
+                            
+                            expect(result).toNot(beNil())
+                            expect(result?.entities.count).to(equal(2))
+                            expect(result?.entities.map({ $0?.name })).to(contain("Victor"))
+                            expect(result?.entities.map({ $0?.name })).to(contain("Hugo"))
+                            expect(result?.errors.count).to(equal(0))
+                            
+                            let entities = kinveyFind(dataStore: networkDataStore).entities
+                            expect(entities).toNot(beNil())
+                            expect(entities?.count).to(equal(2))
+                            expect(entities?.map({ $0.name })).to(contain("Victor"))
+                            expect(entities?.map({ $0.name })).to(contain("Hugo"))
+                        }
+                        it("should sent PUT requests for an array of items with _id") {
+                            let id1 = UUID().uuidString
+                            let id2 = UUID().uuidString
+                            let result = kinveySaveMulti(
+                                dataStore: networkDataStore,
+                                entities: [
+                                    Person { $0.entityId = id1 },
+                                    Person { $0.entityId = id2 }
+                                ]
+                            ).result
+                            
+                            expect(result).toNot(beNil())
+                            expect(result?.entities.count).to(equal(2))
+                            expect(result?.entities.map({ $0?.entityId })).to(contain([id1, id2]))
+                            expect(result?.errors.count).to(equal(0))
+                            
+                            let entities = kinveyFind(dataStore: networkDataStore).entities
+                            expect(entities).toNot(beNil())
+                            expect(entities?.count).to(equal(2))
+                        }
+                        it("should combine POST and PUT requests for items with and without _id") {
+                            let result = kinveySaveMulti(
+                                dataStore: networkDataStore,
+                                entities: [
+                                    Person { $0.entityId = UUID().uuidString },
+                                    Person { $0.entityId = UUID().uuidString },
+                                    Person(),
+                                    Person()
+                                ]
+                            ).result
+                            
+                            expect(result).toNot(beNil())
+                            expect(result?.entities.count).to(equal(4))
```

Comment:
We should make sure the items are returned in the original order

Reply:
done